src/components/CustomProperties/CustomProperties.tsx

```diff
@@ -29,28 +27,20 @@ export function CustomProperties(props: CustomPropertiesProps) {
     colorScheme = DEFAULT_COLOR_SCHEME,
   } = props;
 
-  useEffect(() => {
-    let styleSheet = document.getElementById(STYLE_SHEET_ID);
-
-    if (styleSheet) return;
-
-    styleSheet = document.createElement('style');
-
-    styleSheet.id = STYLE_SHEET_ID;
-    styleSheet.textContent = styles;
-
-    document.head.appendChild(styleSheet);
-  }, []);
-
   return (
-    <Component
-      p-color-scheme={colorScheme}
-      className={className}
-      // TODO: Remove this inline style when we update individual components
-      // to set their own color and background-color properties.
-      style={{color: 'var(--p-text)'}}
-    >
-      {children}
-    </Component>
+    <>
+      <style
+        // Convenience attribute for locating the stylesheet in the DOM.
+        data-polaris-custom-properties=""
+        dangerouslySetInnerHTML={{__html: styles}}
+      />
```

Comment:
Note: Including style tags in the body was officially introduced in HTML 5.2 and is supported in all major browsers (See Make style in body conforming):
https://w3c.github.io/test-results/html52/implementation-report.html